store/commitment/iavl/tree.go

```diff
@@ -83,6 +94,142 @@ func (t *IavlTree) Prune(version uint64) error {
 	return t.tree.DeleteVersionsTo(int64(version))
 }
 
+// Snapshot implements snapshottypes.CommitSnapshotter.
+func (t *IavlTree) Snapshot(version uint64, protoWriter protoio.Writer) error {
+	if version == 0 {
+		return fmt.Errorf("the snapshot version must be greater than 0")
+	}
+
+	latestVersion := t.GetLatestVersion()
+	if version > latestVersion {
+		return fmt.Errorf("the snapshot version %d is greater than the latest version %d", version, latestVersion)
+	}
+
+	tree, err := t.tree.GetImmutable(int64(version))
+	if err != nil {
+		return fmt.Errorf("failed to get immutable tree for version %d: %w", version, err)
+	}
+
+	exporter, err := tree.Export()
```

Comment:
is this aligned with iavlv2?

Reply:
export is the naming in iavl v0 so I kept the same name in iavl v2.  `Export` returns a type `ExportNode` which is a minimal set of fields required to reconstruct the tree when ordereded.